### DISCUSSION TITLE
Configuring Babel Per Sub Directories

### DISCUSSION THREAD

**Discussion by Huxpro** [vercel/next.js#37353] [Help]
It doesn't seem like Next supports configuring Babel at the sub directories level [as Babel documented in its "Monorepos" section](https://babeljs.io/docs/en/config-files#monorepos). 

Notably, It seems like `.babelrc`s from sub-directories never gets pick up, which seems to be a known limitation of the current [babel/loader/get-config](https://github.com/vercel/next.js/blob/25d064f812c3b3152497cc0c04328e52c764e654/packages/next/build/babel/loader/get-config.ts#L28-L31) implementation.

Alternatively, Babel suggests of using [`"overrides"`](https://babeljs.io/docs/en/options#overrides) in a project-wide `babel.config.json` to accomplish  this. But it _seems_ like it also doesn't work, which i'm not sure if it's a Next problem, Babel problem, or my own problem of mis-configuration.

My configuration is roughly:

```
console.log("root babel config load.");

const originalPresets = [["next/babel", {}]];
const myPresets = [["next/babel", { ... }]];

const globalPlugins = ...
const myPlugins = ...

module.exports = {
  presets: globalPresets,
  plugins: globalPlugins,
  overrides: [
    {
      test: ["./sub/**/*"],
      presets: myPresets,
      plugins: myPlugins
    },
  ],
};
```

I can't give a minimal repro yet but the observations are that `globalPlugins` and `globalPresets` were applied globally. However, if `globalPlugins` was not provided, then `myPlugins` will actually be used for files `sub`. I'm not sure what happened here but it's super weird.

**Comment by ijjk** [comment]
Hi, we do have custom loading for the babel config which can be seen here https://github.com/vercel/next.js/blob/5384171bb651ab24398168ac26f76fac37a0f841/packages/next/build/babel/loader/get-config.ts#L172 

We may need to account for the `overrides` config while loading the babel config here which we can investigate.
